### ISSUE TITLE
svg() function prefixes random string to the ID of a symbol within the SVG

### ISSUE THREAD

**Issue by mdominguez** [craftcms/cms#3108]
I'm using an SVG sprite / container which holds a list of <symbols> that each represent a separate SVG (created by Gulp). When trying to add this svg sprite through the svg() function, Craft prefixes a random string to the ID of each symbol inside that SVG. Which means I can't address an exact icon later in my code for usage since I don't know the prefixed ID. [View screenshot](https://cl.ly/t1af)

I assume you guys had reasons to prefix these ID's with random strings (if so, which?), but maybe an extra parameter that prevents this functionality could help me ditching other SVG inline plugins that I now only need to print this SVG sprite.

Glad to hear your thoughts on this!

I've added the SVG sprite too so you can easily test it out => [icons.svg.zip](https://github.com/craftcms/cms/files/2204881/icons.svg.zip)

**Comment by joeydi** [comment]
I'd love to see this issue opened up again.

I agree with @mdominguez that there should be a parameter to disable this behavior.

**Comment by brandonkelly** [comment]
@joeydi did you read my reply? Is there a reason you need it that wouldn’t be covered by including the SVG as a Twig template?

**Comment by joeydi** [comment]
Yeah I did read your reply. Saving the SVG into my templates directory works, but it's not a template. It's an image, and I would prefer it stayed with the rest of my static resources.

This is very confusing, undocumented behavior, and there should be a way to disable it.

**Comment by brandonkelly** [comment]
> Saving the SVG into my templates directory works, but it's not a template. It's an image, and I would prefer it stayed with the rest of my static resources.

SVGs technically aren’t images, it’s a document format just like HTML (which is why they can include JS scripts, etc.), and when you embed an SVG in an HTML file (as opposed to loading it with an `<img>` tag), it becomes part of that HTML document just like a `<p>` node. So at least personally, I think it does make a lot of sense to have embedded SVGs live in the `templates/` folder alongside other HTML include templates.

The only exception is when the SVG is more like site _content_ than part of the site _build_ – e.g. if it is a user-uploaded asset – or comes from an untrusted source. Those are the cases that the `svg()` function is for, and in those cases the `id` namespacing should never be a problem.

But even if you don’t agree with me, the other reason I’d rather not encourgae `svg()` usage when it’s not necessary is that it’s more performant to include them as templates, since Craft won’t have to verify that the file is a real SVG every time, and sanitize it for JS scripts.

> This is very confusing, undocumented behavior

Fair point there. I just added a [note](https://docs.craftcms.com/v3/dev/functions.html#svg-svg-sanitize) about this to the documentation.

**Comment by joeydi** [comment]
Ok. I’m convinced ;) Docs update is appreciated 👍🏼
